src/app/loadout-builder/process-worker/process-utils.ts

```diff
@@ -136,7 +136,7 @@ export function updateMaxTiers(
     // Since we calculate the maximum tier we can hit for a stat in isolation,
     // require all other stats to hit their constrained minimums, but for this
     // stat we start from the highest tier we've observed.
-    const explorationStats = [...requiredMinimumExtraStats];
+    const explorationStats = requiredMinimumExtraStats.slice();
     explorationStats[statIndex] = minMax[statIndex].max * 10 - setStat;
```

Comment:
If you're testing performance, you might want to also try:

```
const explorationStats = requiredMinimumExtraStats.with(statIndex, minMax[statIndex].max * 10 - setStat)
```

Reply:
Microbenchmarks suggest `with` is ever so slightly slower. I'd love to set up some benchmarks that yield tangible results (see #9993).